api/versioninfo.php

```diff
@@ -49,49 +50,18 @@ function remove_utf8_bom($text) {
   
   foreach($platforms as $platform) {
     // Collect all the relevant directories for the platform, ordered by newest release first
-    $dirs = scandir("../$platform");
-    $dirs = array_filter($dirs, 'version_filter');
-    usort($dirs, 'version_compare_backward');
-    
+   
     $p = array();
     
-    foreach($dirs as $dir) {
-      // For each directory
-      $files = glob("../$platform/$dir/{.download_info,*.download_info}", GLOB_NOSORT|GLOB_BRACE);
-      if($files === FALSE || count($files) == 0) continue;
-      for($i = 0; $i < count($files); $i++) {
-        // Read the JSON
-        $data = remove_utf8_bom(file_get_contents("{$files[$i]}"));
-        $json = json_decode($data);
-        if($json === NULL) continue;
-        
-        // Check for stability property
-        if(!property_exists($json, 'stability')) continue;
-        $stability = $json->stability;
-        
-        // Add the property to the platform data
-        if(!array_key_exists($stability, $p)) {
-          $p[$stability] = array("version" => $dir, "path" => "/$platform/$dir");
-        
-          //
-          // If we find a beta before we find an alpha, then the alpha is same as beta, and same as stable
-          // We are only testing 3 labels for now, so let's keep the test simple
-          //
-          
-          if($stability == 'stable') {
-            if(!array_key_exists('beta', $p)) $p['beta'] = $p['stable'];
-          }
-          if($stability == 'stable' || $stability == 'beta') {
-            if(!array_key_exists('alpha', $p)) $p['alpha'] = $p['beta'];
-          }
-        }
-        
-        break;
+    foreach($tiers as $tier) {
+      $dirs = scandir("../$platform/$tier");
+      $dirs = array_filter($dirs, 'version_filter');
+      if(count($dirs) > 0) {
+        usort($dirs, 'version_compare_backward');
+        $p[$tier] = $dirs[0];
```

Comment:
scandir gives alphabetical sort, which doesn't work with version strings (e.g. '9.5' < '10.0' but alphabetical sort doesn't do that)